adapter/internal/messaging/notification_listener.go

```diff
@@ -138,15 +138,36 @@ func handleAPIEvents(data []byte, eventType string) {
 								return
 							}
 						}
-						queryParamMap := make(map[string]string, 3)
-						queryParamMap[eh.GatewayLabelParam] = configuredEnv
-						queryParamMap[eh.ContextParam] = apiEvent.Context
-						queryParamMap[eh.VersionParam] = apiEvent.Version
-						// TODO: (VirajSalaka) Fix the REST API call once the APIM Event hub implementation is fixed.
-						// TODO: (VirajSalaka) Optimize the number of requests sent to /apis endpoint as the same API is returned
-						// repeatedly. (If Eventhub implementation is not fixed)
-						go eh.InvokeService(eh.ApisEndpoint, eh.APIListMap[env], queryParamMap,
-							eh.APIListChannel, 0)
+						var apiEvent APIEvent
+						json.Unmarshal([]byte(string(data)), &apiEvent)
+						api := types.API{
+							APIID:    apiEvent.APIID,
+							UUID:     apiEvent.UUID,
+							Provider: apiEvent.APIProvider,
+							// hardcoded as default version is not assigned at the moment.
+							IsDefaultVersion: false,
+							APIStatus:        apiEvent.APIStatus,
+							TenantID:         apiEvent.TenantID,
```

Comment:
blocking condition's event vs deployevent 's lifecycle status needs to be checked.